### ISSUE TITLE
Automate release process

### ISSUE THREAD

**Issue by armenzg** [armenzg/mozilla_ci_tools#29]
* Change version to final (mozci/__init__.py)
* Merge master to stable
* Tag release
* Push changes
* python setup.py sdist upload
* Bump the version in master and append "-alpha"
* Push new version

Adding the tag helps doing comparisons through github.
We always release from "stable".

**Comment by vaibhavmagarwal** [comment]
We can either build this from scratch or use existing softwares like https://github.com/zestsoftware/zest.releaser/

**Comment by armenzg** [comment]
It sounds that it does the version managing, history tracking and tagging so I'm happy to give it a try!

**Comment by armenzg** [comment]
The following steps need to be verified next time we do a release since I started jumping a bit over all the place. The previous releases I did through GH's UI instead of a mix of cli and UI.

NOTE: I don't see the value of having a "stable" branch.

Steps taken:
* Bump the version to it's final release (mozci/__init__.py)
* Commit with "Release $version"
* Create annotated tag version (git tag -a $version)
** An annotated tag shows better in GH [1]
* Share the tag - (git push origin $version)
* git checkout stable
* git pull (I believe this is to rebase stable - I could be wrong)
* git push
At this point, both branches have the same contents.

Draft a new release through GH's UI [2][3]
Review the changes since the last release [4] and add what is relevant.
e.g. https://github.com/armenzg/mozilla_ci_tools/releases/tag/0.2.4
I wonder if there is anyway to do this through the CLI.

Release to pypi (python setup.py sdist upload)

Start new trunk:
* git checkout master
* Bump version to "-alpha"
* Commit and push

[1] https://github.com/armenzg/mozilla_ci_tools/releases/tag/0.2.4
[2] https://github.com/armenzg/mozilla_ci_tools/releases
[3] https://github.com/armenzg/mozilla_ci_tools/releases/new?tag=0.2.4
[4] https://github.com/armenzg/mozilla_ci_tools/compare/0.2.3...0.2.4

**Comment by vaibhavmagarwal** [comment]
I have figured this out. zest.releaser is a fantastic package to automate the release process directly from commandline!
I would recommend reading their official documentation: http://zestreleaser.readthedocs.org/en/latest/overview.html
Also, see PR https://github.com/armenzg/mozilla_ci_tools/pull/47 for the changes that needs to be done to make this work.
To give an overview with what I tested locally:

1) Download the zest.releaser package from pip: 'pip install zest.releaser'
2) One can do prerelease, release, and postrelease separately. However, I tried 'fullrelease' on commandline and it did all the 3 things just fine.
3) Have a .pypirc locally (which is used to automatically upload on the servers)

PS: I strongly believe that there is no need of 'stable' branch and we can just use master branch for releases. It involves unnecessary merging from master branch and can lead to unwanted error. @armenzg, since you also don't see value in it, lets delete it.  zest.releaser works for one release branch (that should be kept master).

PPS: If you want to see how the commandline commands look in zest.releaser, look for "Making a release with zest.releaser" in http://opensourcehacker.com/2012/08/14/high-quality-automated-package-releases-for-python-with-zest-releaser/
